server/src/unifyfs_request_manager.c

```diff
@@ -683,6 +686,7 @@ int rm_cmd_filesize(
     if (filesize_meta > filesize) {
         filesize = filesize_meta;
     }
```

Comment:
@sandrain , we'll still want to consider the file size field in the metadata even with our new collective.  This field is used in truncate operations, especially for extending a file beyond any extent that was actually written to the file.

Reply:
@adammoody I am a bit confused. Isn't the truncate collective apply the correct size to all extent trees?

Reply:
@sandrain , that's right.  The collective truncate will ensure that the extents do not extend past the end of the file.  However, it's possible that someone truncates the file so that the end of the file falls within a hole, where there is no extent.  On a truncate, we also record the file size in the meta data, and then to compute the file size like for a stat, we take the maximum offset of all extents (via filesize collective) and the maximum of the size value stored in the metadata.  We just need to move this block of code after the collective that gets the max extent offset.

Reply:
It helps to look at the old truncate code here:
https://github.com/LLNL/UnifyFS/blob/ec593f35f36a59ed5d3aeb84dbf053f50674172e/server/src/unifyfs_request_manager.c#L853

Note that it does two things:
1) it deletes/rewrites any extents that extend past the end of the file, in this case by removing/overwriting any keys in MDHIM and
2) it updates the size field in the metadata.

We'll want to do those same two steps here, except that step 1 is now using the collective to update the local extent maps on each server.

Reply:
And to complete the picture, also take a look at the old file size code:
https://github.com/LLNL/UnifyFS/blob/ec593f35f36a59ed5d3aeb84dbf053f50674172e/server/src/unifyfs_request_manager.c#L542
To compute the file size, this does two things:
1) Query all extents in MDHIM to find the max write offset across all extents and
2) query the size field from the file meta data to see if that's even larger

Again in our new collective-based method, we're replacing the query to MDHIM in step 1 with a collective across servers.  However, we still need to do step 2.

Reply:
@adammoody I think I understand your point. thanks.